### PR TITLE
Add documentation

### PR DESCRIPTION
**Pull request by anand2312**
This PR sets up documentation for the project using sphinx and furo, and adds a lot of missing type-hints/docstrings.
The documentation is by no means _complete_ with this, but I think it's a start 😄 

To view the documentation, checkout this branch and run `make html` in the docs directory.

This PR also adds some code changes:
- Reset session.params / headers in the `execute` method, instead of creating a new `httpx.Client` for every request

(I forgot to make that a separate PR, sorry 😅 )

### CODE REVIEW

**Review by J0 — CHANGES_REQUESTED** [review]
Lorn, thanks for putting this together!

**Review by J0 — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -136,3 +136,6 @@ dmypy.json
 
 # Poetry local config
 poetry.toml
+
+# dummy test file
```
Comment (J0) [inline .gitignore:140]: Is this still relevant?

Comment (anand2312) [reply .gitignore]: Whoops, didn't mean to leave that in. Good catch, reverted in [300a089](https://github.com/supabase-community/postgrest-py/pull/111/commits/300a089ac19bcb0b1cbfec13a7429b46ca51f5ed)

**Comment by anand2312** [comment]
@J0 I tried deploying to RTD but I seem to not have enough permissions on this repo (it needs permissions to manage webhooks). Can you try?

**Comment by J0** [comment]
Weird, I'll look into getting you those permissions. One sec.

**Comment by anand2312** [comment]
@J0 the deployed version is pointing to my fork right now, not this repo. I did that because I couldn't get the config files just right so I needed to test first 😅 I'll move it here in a bit once I get the permissions. Or you could send me your readthedocs account name and I'll add you as a maintainer, and then you could set up the webhook (the webhook is to automatically deploy updated documentation when new commits are made)

**Comment by anand2312** [comment]
Alternatively, we can deploy docs to Github Pages intead of readthedocs. The domain name would then be `https://supabase-community.github.io/postgrest-py/`
Pro: having everything coupled to this repo would be convenient
Con: Readthedocs offers versioned docs by default which could be quite nice

**Pull request merged**